lib/resty/iprepd.lua

```diff
@@ -40,7 +40,20 @@ function _M.new(options)
     whitelist = iputils.parse_cidrs(whitelist_list)
   end
 
+  --[[ Initialize an http client with a timeout of 50ms by default
+       that is set to keepalive and stay idle for up to 2 minutes.
+       Each nginx worker will keep up to 10 of these clients in its pool. ]]
+  local httpc = http.new()
+  httpc:set_timeout((options.timeout or 50), (options.timeout or 50), (options.timeout or 50))
+  ok, err = httpc:set_keepalive((options.http_max_idle_timeout or 120000), (options.http_pool_size or 10))
```

Comment:
I'm not sure if this is correct use of this function here

https://github.com/ledgetech/lua-resty-http#set_keepalive

```
Only call this method in the place you would have called the close method instead.
Calling this method will immediately turn the current http object into the closed state.
Any subsequent operations other than connect() on the current object will return the closed error.
```

Should this instead be called at some point after the call to `request_uri`?

Since this causes the `httpc` to be in a closed state does this mean we need to call `http.new` for each request?

`request_uri` appears to actually call `set_keepalive` when the request is finished.

Reply:
I read this initially and was confused by it. It also doesn't match the tests: https://github.com/ledgetech/lua-resty-http/blob/master/t/07-keepalive.t#L160